includes/Integrations/AMP.php

```diff
@@ -162,8 +163,15 @@ public function add_amp_content_sanitizers( $sanitizers ) {
 
 		$story = new Story();
 		$story->load_from_post( $post );
+
+		$publisher_logo_id  = get_post_meta( $post->ID, Story_Post_Type::PUBLISHER_LOGO_META_KEY, true );
+		$publisher_logo_url = wp_get_attachment_image_url( $publisher_logo_id, Image_Sizes::PUBLISHER_LOGO_IMAGE_SIZE );
+
+		/* This filter is documented in includes/Discovery.php */
+		$publisher_logo_url = apply_filters( 'web_stories_publisher_logo', $publisher_logo_url, $post );
```

Comment:
This logic is repeated a couple of times. Let's create a method in the publisher trait. Something like this. 

```
   public function get_story_publisher_logo( WP_Post $post ){
          $publisher_logo_id  = get_post_meta( $post->ID, Story_Post_Type::PUBLISHER_LOGO_META_KEY, true );
		  $publisher_logo_url = wp_get_attachment_image_url( $publisher_logo_id, Image_Sizes::PUBLISHER_LOGO_IMAGE_SIZE );
          /**
		  * Filters the publisher logo URL.
		  *
		  * @since 1.0.0
		  * @since 1.1.0 The second parameter was deprecated.
		  * @since 1.11.0 The second parameter was repurposed to provide the current post object.
		  *
		  * @param string|null $logo_image_url Publisher logo URL.
		  * @param WP_Post     $post           Current post object.
		 */
		return apply_filters( 'web_stories_publisher_logo', $url, $post );
  }
```

Reply:
Would be specially usefully https://github.com/google/web-stories-wp/issues/7846

Reply:
It's not always the same usage. For example, in `Discovery.php` the publisher logo width & height are used as well. The URL alone is needed in 2 places outside of that.
IMO not a big deal if that's repeated like that.

Reply:
But all the publisher replated code is meant to in that one place. Why remove it out, making the code less readable and discovable?

Reply:
Also think that code that is reused twice, should have a unit test.

Reply:
Due to different usages that would probably mean adding multiple different methods (`get_publisher_logo_id( $post )`, `get_publisher_logo_id( $post )`, `get_publisher_logo_url( $post )`, `get_publisher_logo_src( $post )`), making this a lot more complex and less comprehensible imo.

What about moving this code to the `Story` model instead? Since the logo is tied to a story anyway.

Reply:
> What about moving this code to the Story model instead? Since the logo is tied to a story anyway.

Love this idea! Can't believe I didn't think of it is!

So two new properties? 
`publisher_logo_url` - string / url
`publisher_logo_size` - array, [width, height]

Reply:
tried this now